File: tools/commit-benchmark/mandelbrot.php
<?php
require getcwd().'/php/src/bootstrap.php';
$source=file_get_contents('examples/mandelbrot.sel');
$t=hrtime(true);$program=\Sel\Sel::compile($source);$compile_ms=(hrtime(true)-$t)/1e6;
$samples_ms=[];$outputs=[];
$warmups=(int)(getenv('MANDEL_WARMUPS') !== false ? getenv('MANDEL_WARMUPS') : 2);$runs=(int)(getenv('MANDEL_RUNS') ?: 5);
for($i=0;$i<$warmups+$runs;$i++){$t=hrtime(true);$result=$program->run(\Sel\Value::none());$output=$result->asText();$elapsed=(hrtime(true)-$t)/1e6;if($i>=$warmups){$samples_ms[]=$elapsed;$outputs[]=$output;}}
file_put_contents($argv[1],json_encode(compact('compile_ms','samples_ms','outputs','warmups','runs')));

File: tools/commit-benchmark/mandelbrot.py
import json, sys, time, os
from pathlib import Path
sys.path.insert(0, str(Path.cwd() / 'python'))
from sel import compile, Value
source = Path('examples/mandelbrot.sel').read_text()
t = time.perf_counter(); program = compile(source); compile_ms = (time.perf_counter()-t)*1000
samples=[]; outputs=[]
warmups=int(os.environ.get('MANDEL_WARMUPS','2')); runs=int(os.environ.get('MANDEL_RUNS','5'))
for i in range(warmups+runs):
    t=time.perf_counter(); result=program.run(Value.none()); output=result.as_text(); elapsed=(time.perf_counter()-t)*1000
    if i>=warmups: samples.append(elapsed); outputs.append(output)
Path(sys.argv[1]).write_text(json.dumps(dict(compile_ms=compile_ms,samples_ms=samples,outputs=outputs,warmups=warmups,runs=runs)))
